### ISSUE TITLE
Argument 1 passed to Honeybadger\Honeybadger::setAction() must be of the type string, null given

### ISSUE THREAD

**Issue by drbyte** [honeybadger-io/honeybadger-laravel#34]
## What are the steps to reproduce this issue?

1. Install 1.7.0 of this package, on Laravel 5.8.35
2. Visit a bad URL on the site

## What happens?
500 error, and log message:
```
Argument 1 passed to Honeybadger\Honeybadger::setAction() 
must be of the type string, null given, 
called in /vendor/honeybadger-io/honeybadger-laravel/src/Middleware/HoneybadgerContext.php
 on line 47
```

## What versions are you using?
**Package Version:** 1.7.0

Seems to be related to https://github.com/honeybadger-io/honeybadger-laravel/pull/31

**Comment by jessarcher** [comment]
I just ran into this on a Laravel 6.0 app.

In my case, it's because I have a route using a closure.

The `HoneybadgerContext` middleware is expecting `Route::getCurrentRoute()->getActionName())` to return something in the form of `ControllerName@action`, however for a closure, it simply returns `Closure`. The `explode('@'...` call only returns a single item, and so a null coalesce causes `setAction()` to receive the unwanted `null`.

I'm not sure what *should* happen in this situation as I don't know enough about what the Honeybadger API will expect, but I'd imagine we'd want to either:
* Pass _something_ to `setAction()`, like perhaps the string 'Closure' which is already being passed to `setComponent()`
* Don't call `setAction()` if we don't have anything to pass - I'm not sure whether calling `setComponent()` without `setAction()` is okay though.
* Modify `setAction()` to accept null - again, not sure if this is okay.

**Comment by sixlive** [comment]
EEK! I’ll address this first thing tomorrow!

**Comment by drbyte** [comment]
Thanks for pointing out the closure aspect!

In case it helps I've seen it on closures for two different scenarios:

```php
Route::fallback(function () {
    return response()->view('errors.404', request(), 404);
})->name('fallback');
```

```php
Route::domain('spam.example.com')->group(function () {
    Route::any('{any}', function () {
        abort(404);
    })->where('any', '.*');
});
```

**Comment by sixlive** [comment]
Just opened #35 to resolve this issue. Will tag as soon as CI is complete.

**Comment by sixlive** [comment]
@drbyte @jessarcher gosh, really sorry about this one shouldn't have happened. Just tagged the fix in [1.7.1](https://github.com/honeybadger-io/honeybadger-laravel/releases/tag/v1.7.1)

**Comment by drbyte** [comment]
Thanks for the quick fix!
